content/docs/esc/other-integrations/cloudflare.md

```diff
@@ -0,0 +1,254 @@
+
+```bash
+esc env edit ${ESC_ENV}
+```
+
+Paste the contents below in the editor and replace the values. Note: the API token is declared as a secret. Once the Environment is saved, Pulumi will encrypt its value and replace it with ciphertext.
+
+```yaml
+values:
+  environmentVariables:
+    CLOUDFLARE_API_TOKEN:
+      fn::secret: abc123abc123abc123abc123abc123
+    CLOUDFLARE_ACCOUNT_ID: abc123abc123
+```
+
+Now that the Pulumi ESC Environment is created, it can be consumed in a variety of ways, such as running other shell commands without having to set the environment variables locally first.
+
+The `esc run` command opens the Environment you previously created, sets the specified environment variables, and then uses those environment variables in the context of the `wrangler` commands.
+
+### 2. Use ESC with `wrangler whoami`
+
+Log into your Cloudflare account without needing to manage the credentials directly in your shell:
+
+```bash
+# ensure you're currently not logged in
+npx wrangler logout
+
+# retrieve esc environment and authenticate programmatically
+esc run ${ESC_ENV} npx wrangler whoami
+```
+
+Because we are running `wrangler` in non-interactive mode, it requires a Cloudflare API token and account ID for authentication. The `wrangler whoami` retrieves details about the provided credentials.
+
+For additional options and details, see `esc run --help`.
+
+### 3. Use ESC to create your `.dev.vars` file
+
+The `.dev.vars` file is located in the root of your wrangler project to define secrets used when running `wrangler dev`.
+
+{{< notes type="info" >}}
+**Recommendation** Use a dedicated ESC Environment for your `.dev.vars` definitions because the esc `dotnev` format flag does not support property paths.
+{{< /notes >}}
+
+Create a new ESC Environment:
+
+```bash
+ESC_ENV=my-dev-vars-environment
+esc env init ${ESC_ENV}
+```
+
+Edit the environment in your terminal:
+
+```bash
+esc env edit ${ESC_ENV}
+```
+
+Paste the contents below in the editor:
+
+```yaml
+values:
+  environmentVariables:
+    TOP_SECRET:
+      fn::secret: "the moon is made of cheese"
+```
+
+Per [Cloudflare documentation](https://developers.cloudflare.com/workers/configuration/secrets/#secrets-in-development), the `.dev.vars` file should be formatted like a `dotenv` file.
+
+```bash
+# generate the `.dev.vars` file:
+esc env open ${ESC_ENV} --format dotenv > .dev.vars
+
+# verify the contents
+cat .dev.vars
+```
+
+### 4. Use ESC with `wrangler secret put`
+
+With Pulumi ESC you can centralize common secrets and then use Wrangler to pass them on to your Workers and other Cloudflare resources as needed:
+
+Add a new value to your `my-dev-environment` Environment:
+
+```bash
+ESC_ENV=my-dev-environment
+esc env set ${ESC_ENV} environementVariables.TOP_SECRET "aliens are real" --secret
+```
+
+Share the secret with your Worker.
+
+```bash
+esc run -i ${ESC_ENV} -- sh -c 'echo "$TOP_SECRET" | npx wrangler secret put TOP_SECRET'
```

Comment:
I'm not sure if we explained the -i interactive mode or if we need to

Reply:
added a note.